src/App.tsx

```diff
@@ -88,6 +88,10 @@ function App(props: any) {
   useEffect(() => {
     if (selectedNetwork && selectedNetwork.name !== query.network) {
       setQuery({ network: selectedNetwork.name });
+      history.push({
+        pathname: history.location.pathname,
+        search: `?network=${selectedNetwork.name}`,
```

Comment:
Will this work if there are other parameters in the search pattern of the URI?

Reply:
it does for certain keys, see here: https://github.com/etclabscore/expedition/blob/master/src/App.tsx#L38

Reply:
Gotcha, thanks. Maybe the supported params should be parts of a config object and standardized/globalized then too, toward fewer magic strings.

Would suggest new ticket for it.

Reply:
... Or modify the handrolled `History` logic to use either a library (there _must be one_...?) and/or/thereby support arbitrary URI path/parameter/hash updates.